### PR TITLE
Allow IP blacklisting of b2b traffic

### PR DESCRIPTION
**Pull request by commuterjoy**
I'm not sure of the benefit of this. I'll run it for a few days and see what impact it has, then discuss with product about keeping it.

### CODE REVIEW

File: server/lib/b2b.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+
+'use strict';
+
+const Poller = require('ft-poller');
+const aws4 = require('aws4');
+const debug = require('debug')('lookup');
+const hostname = 'ft-next-redshift.s3.amazonaws.com';
+let b = { '1.1.1.1': 1 }; // test fixture
+
+const signed = aws4.sign({
```
Comment (matthew-andrews) [inline server/lib/b2b.js]: neat

File: test/server/forced-opt-in.spec.js [hunk 2]
```diff
@@ -117,6 +117,15 @@ describe('Anonymous Opt In', function(){
 				.expect(302, done);
 		});
 
+		it('Should send b2b traffic back to ft.com', function(done){
```
Comment (matthew-andrews) [inline test/server/forced-opt-in.spec.js]: this is kind of a shame :-(

File: Procfile [hunk 3]
```diff
@@ -1 +1 @@
-web: ./node_modules/.bin/forever ./server/app.js;
+web: ./node_modules/.bin/forever -c 'node' ./server/app.js
```
Comment (wheresrhys) [inline Procfile]: what this do?

File: test/server/forced-opt-in.spec.js (hunk 2 above)
Comment (wheresrhys) [inline test/server/forced-opt-in.spec.js]: Could AB test this. Are people on b2b IPs more likely to sign in/register if they see next?

Comment (matthew-andrews) [inline test/server/forced-opt-in.spec.js]: :green_heart:

File: Procfile (hunk 3 above)
Comment (commuterjoy) [inline Procfile]: oops. mistake.

File: test/server/forced-opt-in.spec.js (hunk 2 above)
Comment (commuterjoy) [inline test/server/forced-opt-in.spec.js]: I think it's mostly FUD that caused this but I'd rather do it and move to 10% than debate it for a week, so I plan to look at the stats next week and see what impact it's having. I said to product this code is going to get out of sync quickly (as ip addresses changes week to week) so they have ~3/4 weeks to do any comms with b2b before this goes. Will do a feature flag to remind myself.

Comment (commuterjoy) [inline test/server/forced-opt-in.spec.js]: this is made much harder by there being no sensible way to distinguish b2b and b2c in the stats :(

**Pull request merged**